app/modules/import-export/module.php

```diff
@@ -541,7 +552,11 @@ private function maybe_handle_ajax() {
 			}
 		} catch ( \Error $e ) {
 			if ( isset( $this->import ) ) {
-				$this->import->finalize_import_session_option();
+				$is_third_party_plugin_error = $this->is_third_party_plugin_class( $e->getTrace()[0]['class'] );
+
+				$is_third_party_plugin_error ?
+					wp_send_json_error( self::THIRD_PARTY_PLUGIN_ERROR, 500 ) :
+					$this->import->finalize_import_session_option();
```

Comment:
```suggestion
				if ( ! $this->is_native_class( $e->getTrace()[0]['class'] ) ) {
					wp_send_json_error( self::THIRD_PARTY_PLUGIN_ERROR, 500 );
				}

				$this->import->finalize_import_session_option() :
```

Reply:
Also, I'm not sure this is the right place to add the check. As basically we want to know which runner failed with the error IMO. Let's talk about it?